components/applications-service/pkg/storage/postgres/hab_event.go

```diff
@@ -134,41 +134,167 @@ func (db *Postgres) IngestHealthCheckEventWithoutMetrics(event *habitat.HealthCh
 		pkgIdent,
-		strings.ToUpper(event.GetResult().String()),
+		eventHealth,
 	)
 
-	if err1 != nil {
+	if err != nil {
 		// We will retry once if a unique constraint is hit,
 		// in case a new deployment or service group is in the same batch of events.
-		if err1.Error() == uniqueDeploymentError || err1.Error() == uniqueServiceGroupError {
+		if err.Error() == uniqueDeploymentError || err.Error() == uniqueServiceGroupError {
 			return db.insertNewService(
 				eventMetadata,
 				svcMetadata,
 				pkgIdent,
-				strings.ToUpper(event.GetResult().String()),
+				eventHealth,
 			)
 		}
 	}
-	return err1
+	return err
+}
+
+func (db *Postgres) updateTables(
+	svc *service,
+	eventMetadata *habitat.EventMetadata,
+	svcMetadata *habitat.ServiceMetadata,
+	pkgIdent *packageIdent,
+	health string) error {
+
+	//updates := map[string]interface{}{}
+	db.updateService(svc, eventMetadata, svcMetadata, pkgIdent, health)
+
+	deploy, err := db.getDeployment(svc.DeploymentID)
+	if err != nil {
+		return errors.Wrap(err, "unable to update tables")
+	}
+	db.updateDeployment(deploy, eventMetadata)
+
+	sup, err := db.getSupervisor(svc.SupID)
+	if err != nil {
+		return errors.Wrap(err, "unable to update tables")
+	}
+	db.updateSupervisor(sup, eventMetadata)
+
+	sg, err := db.getServiceGroup(svc.GroupID)
+	if err != nil {
+		return errors.Wrap(err, "unable to update tables")
+	}
+	db.updateServiceGroup(sg, svcMetadata)
+
+	return db.triggerDataUpdates(svc, deploy, sup, sg)
+}
+
+func (db *Postgres) triggerDataUpdates(
+	svc *service,
+	deploy *deployment,
+	sup *supervisor,
+	sg *serviceGroup) error {
+
+	// update only if there is something to update
+	return dblib.Transaction(db.DbMap, func(tx *gorp.Transaction) error {
+
+		if svc.needUpdate {
+			if _, err := tx.Update(svc); err != nil {
+				return errors.Wrap(err, "unable to update service")
+			}
+		}
+
+		if deploy.needUpdate {
+			if _, err := tx.Update(deploy); err != nil {
+				return errors.Wrap(err, "unable to update deployment")
+			}
+		}
+
+		if sup.needUpdate {
+			if _, err := tx.Update(sup); err != nil {
+				return errors.Wrap(err, "unable to update supervisor")
+			}
+		}
+
+		if sg.needUpdate {
+			if _, err := tx.Update(sg); err != nil {
+				return errors.Wrap(err, "unable to update service group")
+			}
+		}
+
+		return nil
+	})
+}
+
+// updates the provided supervisor from a HealthCheck event
+func (db *Postgres) updateSupervisor(
+	sup *supervisor, eventMetadata *habitat.EventMetadata) {
+
+	if sup.Fqdn != eventMetadata.GetFqdn() {
+		sup.Fqdn = eventMetadata.GetFqdn()
+		sup.needUpdate = true
+	}
+
+	if sup.Site != eventMetadata.GetSite() {
+		sup.Site = eventMetadata.GetSite()
+		sup.needUpdate = true
+	}
+
+	// TODO @afiune we could have this column in all tables as well
+	// when a supervisor needs update, update the timestamp of the last event received
+	//if sup.needUpdate {
+	//sup.LastEventOccurredAt = convertOrCreateTimestamp(eventMetadata.GetOccurredAt())
+	//}
```

Comment:
I'm not sure what the value would be here, do we need to leave this TODO? Are we gonna do it?

Reply:
I will remove it, what I wanted to do is to have the same field we have in the `service` table to know "What was the last Habitat message that modified `XYZ` table"?

You could know things like; oh look, the supervisor (eg. FQDN of the machine) was modified with a Habitat message that occurred at this time.

If we need this we can talk about it in standup and apply it, just for the record, we won't be saving the timestamp of hab messages unless the message modifies a field that belongs to the `service` table.